lib/gen_stage.ex

```diff
@@ -553,12 +553,12 @@ defmodule GenStage do
   of the demand-driven protocol but respecting the event ordering.
   See `sync_notify/3` and `async_notify/2`.
 
-  Notifications are useful for out-of-band information, for example,
+  Notifications are useful for out-of-band information: for example,
```

Comment:
I think the colon is quite appropriate here as we are explaining what we said in the first sentence. `.` is still better than `,` in my opinion, as a pair of `,`s (like in ", for example,") is similar to parenthesis, which are omittable without changing the meaning of the sentence, but here if you remove the "for example" the sentence sounds really weird (because I think it's two sentences. What about `;`? :)